### ISSUE TITLE
Ignore folders and file patterns

### ISSUE THREAD

**Issue by ShadowJonathan** [TagStudioDev/TagStudio#14]
Sorta like a `.gitignore`, to be able to ignore folder patterns like (ironically) `.git`, cache folders, and .DS_Store and the likes.

Else all the hundreds or so hidden files will clog up the interface, with no way to hide them.

**Comment by shadowking123** [comment]
i have implemented this feature by entering the "!" in the start of the query and then the file name you want to filter. 
if the file name is in the query, remove it from search.
should it have more than that?

**Comment by ShadowJonathan** [comment]
yes, it should be more-or-less permanent, as a "ignore all these folders all the time" kind of setting, a permanent file/folder blacklist

**Comment by shadowking123** [comment]
How no i get it.
I will work on it

**Comment by Loran425** [comment]
Hiding files with specific extensions was added in https://github.com/TagStudioDev/TagStudio/commit/31f402289565b55d3a84d11f0eb6055e9bf42374

Ignoring specific folders within a library is currently still open.

**Comment by ItsThatOneJack-Dev** [comment]
This would be an amazing feature, and would be useful for many things.

For example, I use a file syncing program called Syncthing to sync my projects and memes folder across devices, and it creates a hidden folder in every synced directory that contains text files with IDs corresponding to files and directories that it uses to ensure files are up-to-date.

It would be great if I could set TagStudio to ignore all hidden directories and files or just add this folder to an ignore list, then TagStudio wouldn't be cluttered by Syncthing files.

**Comment by yedpodtrzitko** [comment]
Currently the list of excluded/allowed file extensions is managed via the GUI. That should remain as is, as not every user is willing/capable to manage that via `.gitignore`-like file.

Having combination of "`.gitignore`" and the GUI interface could have conflicting rules, so I wouldnt go that way of supporting both. 

However there could be an option to have some kind of `.noindex` file (that name is actually taken, so probably something like `.tsnoindex` instead), which would serve as a hard-stop for TS to totally ignore content of given folder. So in the case of syncthing folder, if there would be a file `syncthing/.tsnoindex`, then it would skip that folder.

**Comment by AwesomeParley** [comment]
> However there could be an option to have some kind of `.noindex` file (that name is actually taken, so probably something like `.tsnoindex` instead), which would serve as a hard-stop for TS to totally ignore content of given folder. So in the case of syncthing folder, if there would be a file `syncthing/.tsnoindex`, then it would skip that folder.

I feel like having a ignore folder list which is like the ignore extension list would be way more useful than having to make a folder called `.tsnoindex` or `.noindex`. Although a cool idea, it would mess up a lot of people's workflow with folders. TagStudio, especially in it's current state is helpful for finding videos, pictures, and audio and then we can easily open or get the file location of the item. But sometimes I have folders and projects that I just don't want to see in TagStudio because of the nature of the items in those. 

Ex: I don't want to see my Godot Projects folder because all the items inside are strictly for Godot, and I can't really move those items without breaking the Projects. Same with Blender, and a whole lot of other programs. They are just not designed to be like that. I also don't want to have to go to each individual folder and add it to my Library. 

TagStudio isn't supposed to completely replace folders, it just makes it a whole lot easier to find files in our already created web of folders. Anyways, back to the main topic at hand, having to add a folder called something specific would not be beneficial to the end user. 

Very cool idea, maybe if it's added we could have a toggle for people in the settings to use that.

Adding onto that idea; what you could do is add a way to add any specifically named folders in any folder in the library. Ex: Whenever you are using Blender to do some video editing, if you add a video into the Video Sequence Editor, Blender makes a file called `BL_proxy` which is quite annoying when it adds all of it's `.avi` and `.blend_tc`/`.blen_tc` files into it. I do sometimes need to use `.avi` files when video editing, but if I am blocking all `.avi` files in TagStudio, it's not very helpful when I try to find them again. I'm certain that others have many issues that are also like this. 

Obviously this also adds functionality to add blocking any user created `.tsnoindex` folders as well. Maybe the option could have that as a placeholder?
